src/ml/neural_net/mps_graph_trainer.mm

```diff
@@ -62,17 +63,17 @@ int TCMPSDeleteGraphModule(MPSHandle handle) {
 EXPORT int TCMPSTrainStyleTransferGraph(MPSHandle handle, int index, TCMPSFloatArrayRef inputs,
                                         TCMPSFloatArrayRef labels, TCMPSFloatArrayRef* loss_out) {
   API_BEGIN();
-  mps_graph_cnn_module *obj = (mps_graph_cnn_module *)handle;
+
+  style_transfer *obj = reinterpret_cast<style_transfer *>(handle);
   float_array* inputs_ptr = reinterpret_cast<float_array*>(inputs);
   float_array* labels_ptr = reinterpret_cast<float_array*>(labels);
-  float_scalar* index_ptr = new float_scalar(index);
 
   shared_float_array inputs_array(
       std::make_shared<external_float_array>(*inputs_ptr));
   shared_float_array labels_array(
       std::make_shared<external_float_array>(*labels_ptr));
   shared_float_array index_array(
-      std::make_shared<external_float_array>(*index_ptr));
+      std::make_shared<external_float_array>(float_scalar(index)));
```

Comment:
Like the previous memory management error, this constructs an `external_float_array` that stores pointers into a temporary objects, the `float_scalar`, which will be destroyed before the `external_float_array` is. Please read the comments documenting `external_float_array`.

This line should just be
```
shared_float_array index_array = std::make_shared<float_scalar>(index);
```

Reply:
This memory stuff is kinda tricky, I'll have to understand it. Thanks!